packages/oasis-actions/src/helpers/flashloan/resolve-provider.ts

```diff
@@ -0,0 +1,30 @@
+import { providers } from 'ethers'
+
+import { Network, NetworkByChainId } from '../../../../../helpers/network'
+import { FlashloanProvider } from '../../types/common'
+
+export const resolveFlashloanProvider = async (provider: providers.Provider) => {
+  const chainId = (await provider.getNetwork()).chainId
+
+  if (NetworkByChainId[chainId] === Network.LOCAL) {
+    const localProvider = provider as providers.JsonRpcProvider
+    if (!(localProvider as providers.JsonRpcProvider).send)
+      throw new Error('Provider does not support send method')
+    const metadata = await localProvider.send('hardhat_metadata', [])
+    return resolveProvider(NetworkByChainId[metadata.forkedNetwork.chainId])
+  }
+
+  return resolveProvider(NetworkByChainId[chainId])
+}
+
+function resolveProvider(network: Network): FlashloanProvider {
```

Comment:
All you need this is this function renamed to `resolveFlashloanProvider`

Reply:
We do need this because when running a local node you might be forking either mainnet or optimism.

So, returning just DSSFlash for local node would be insufficient.

Reply:
My point is that you already have more or less the same code in `network.ts`. There should be one place responsible for the network resolution.